server/lyft/gateway/events_controller.go

```diff
@@ -0,0 +1,287 @@
+		scope.Counter(fmt.Sprintf("error_%d", resp.err.code)).Inc(1)
+		w.WriteHeader(resp.err.code)
+		fmt.Fprintln(w, resp.err.err.Error())
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprintln(w, resp.body)
+}
+
+// HandleGithubCommentEvent handles comment events from GitHub where Atlantis
+// commands can come from. It's exported to make testing easier.
+func (g *VCSEventsController) HandleGithubCommentEvent(event *github.IssueCommentEvent, githubReqID string, logger logging.SimpleLogging, r *http.Request) HttpResponse {
+	if event.GetAction() != "created" {
+		return HttpResponse{
+			body: fmt.Sprintf("Ignoring comment event since action was not created %s", githubReqID),
+		}
+	}
+	baseRepo, _, pullNum, err := g.Parser.ParseGithubIssueCommentEvent(event)
+	wrapped := errors.Wrapf(err, "Failed parsing event: %s", githubReqID)
+	if err != nil {
+		return HttpResponse{
+			body: wrapped.Error(),
+			err: HttpError{
+				code: http.StatusBadRequest,
+				err:  wrapped,
+			},
+		}
+	}
+	// We pass in nil for maybeHeadRepo because the head repo data isn't
+	// available in the GithubIssueComment event.
+	return g.handleCommentEvent(logger, baseRepo, pullNum, event.Comment.GetBody(), models.Github, r)
+}
+
+func (g *VCSEventsController) handleCommentEvent(logger logging.SimpleLogging, baseRepo models.Repo, pullNum int, comment string, vcsHost models.VCSHostType, r *http.Request) HttpResponse {
+	parseResult := g.CommentParser.Parse(comment, vcsHost)
+	if parseResult.Ignore {
+		truncated := comment
+		truncateLen := 40
+		if len(truncated) > truncateLen {
+			truncated = comment[:truncateLen] + "..."
+		}
+		return HttpResponse{
+			body: fmt.Sprintf("Ignoring non-command comment: %q", truncated),
+		}
+	}
+	logger.Info("parsed comment as %s", parseResult.Command)
+
+	// At this point we know it's a command we're not supposed to ignore, so now
+	// we check if this repo is allowed to run commands in the first place.
+	if !g.RepoAllowlistChecker.IsAllowlisted(baseRepo.FullName, baseRepo.VCSHost.Hostname) {
+		g.commentNotAllowlisted(baseRepo, pullNum)
+
+		err := errors.New("Repo not allowlisted")
+		return HttpResponse{
+			body: err.Error(),
+			err: HttpError{
+				err:  err,
+				code: http.StatusForbidden,
+			},
+		}
+	}
+
+	// If the command isn't valid or doesn't require processing, ex.
+	// "atlantis help" then we just comment back immediately.
+	// We do this here rather than earlier because we need access to the pull
+	// variable to comment back on the pull request.
+	if parseResult.CommentResponse != "" {
+		if err := g.VCSClient.CreateComment(baseRepo, pullNum, parseResult.CommentResponse, ""); err != nil {
+			logger.Err("unable to comment on pull request: %s", err)
+		}
+		return HttpResponse{
+			body: "Commenting back on pull request",
+		}
+	}
+	logger.Debug("executing command")
+	if err := g.SendToWorker(r); err != nil {
+		logger.With("err", err).Warn("failed to send comment request to Atlantis worker")
```

Comment:
how come you're logging the error as a warning? this should definitely be an error since we're dropping requests.